### PR TITLE
Add customized renormalized moe routing kernel for moe cutlass backend

### PR DESCRIPTION
**Pull request by ChristinaZ**
# Add customized renormalized moe routing kernel for moe cutlass backend
## Description
- Add pytorch op `torch.ops.trtllm.renorm_moe_routing_op` and use it when `num_experts<=128 and num_experts_per_token <= 8`.   
  Related files: cpp/tensorrt_llm/thop/renormMoeRoutingOp.cpp 
                          tensorrt_llm/_torch/modules/fused_moe/routing.py
(Hi @hlu1, since this modification is related to the MOE part. Please help review it. ）

- Add the customized kernel `renormMoeRoutingKernel`
 Related files: cpp/tensorrt_llm/kernels/renormMoeRoutingKernels.cu 
                        cpp/tensorrt_llm/kernels/renormMoeRoutingKernels.cu
- Add unit test test_customized_renormalize_moe_routing()
 Related file:  tests/unittest/_torch/modules/test_moe_routing.py
 
- Besides adding this feature, I also did a small modification for `RoutingMethodType`. By replacing the variable name `RoutingMethodType.Qwen3` with `RoutingMethodType.RenormalizeNaive.`, we hope to facilitate broader usage in the future. 
(Hi @djns99 and @rosenrodt , please help review this part.)

## Test Coverage
```
pytest -k test_customized_renormalize_moe_routing tests/unittest/_torch/modules/test_moe_routing.py
```

### CODE REVIEW

**Review by hlu1 — COMMENTED** [review]
Please make sure to run the multi-gpu tests to test accuracy of qwen3

File: tensorrt_llm/_torch/modules/fused_moe/routing.py [hunk 1]
```diff
@@ -241,7 +252,7 @@ def apply(self,
         return balanced_indices, balanced_values
 
 
-class Qwen3MoeRoutingMethod(BaseMoeRoutingMethod):
+class RenormalizeNaiveMoeRoutingMethod(BaseMoeRoutingMethod):
```
[the remark this answers is not in the document]
Comment (djns99) [reply tensorrt_llm/_torch/modules/fused_moe/routing.py:255]: Yes agree on both of these suggestions. Also add the comment to the enum

File: cpp/tensorrt_llm/thop/renormMoeRoutingOp.cpp [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+    switch (data_type)
+    {
+    case torch::kFloat32:
+        // Handle Float32
+        tk::invokeRenormMoeRouting<float, float, int32_t>(reinterpret_cast<float*>(router_logits.mutable_data_ptr()),
+            reinterpret_cast<float*>(topk_values.mutable_data_ptr()),
+            reinterpret_cast<int32_t*>(topk_indices.mutable_data_ptr()), num_tokens, num_experts, topk, stream);
+        break;
+    case torch::kBFloat16:
```
Comment (djns99) [inline cpp/tensorrt_llm/thop/renormMoeRoutingOp.cpp:51]: Any reason we dont have `half` dtype? `throw std::invalid_argument("Invalid dtype, only supports float16, float32, and bfloat16");` seems to indicate it should be supported

**Review by djns99 — APPROVED** [review]
LGTM, thanks @ChristinaZ

File: tensorrt_llm/_torch/modules/fused_moe/routing.py [hunk 3]
```diff
@@ -86,18 +86,29 @@ class RenormalizeMoeRoutingMethod(BaseMoeRoutingMethod):
...
+            self, router_logits: torch.Tensor) -> (torch.Tensor, torch.Tensor):
         topk_values, topk_indices = torch.topk(router_logits,
                                                k=self.top_k,
                                                dim=-1)
         return topk_indices.to(torch.int32), torch.nn.functional.softmax(
             topk_values.float(), dim=-1)
 
+    def apply(self,
+              router_logits: torch.Tensor) -> (torch.Tensor, torch.Tensor):
+        num_experts = router_logits.shape[1]
```
[the remark this answers is not in the document]
Comment (ChristinaZ) [reply tensorrt_llm/_torch/modules/fused_moe/routing.py]: Thanks, Done.

https://github.com/ChristinaZ/TensorRT-LLM/blob/414a4c2341f5ce45cc747150a9b094a7d224cae2/tensorrt_llm/_torch/modules/fused_moe/routing.py#L105

File: tensorrt_llm/_torch/modules/fused_moe/routing.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ChristinaZ) [reply tensorrt_llm/_torch/modules/fused_moe/routing.py]: Agree. Done.
https://github.com/ChristinaZ/TensorRT-LLM/blob/414a4c2341f5ce45cc747150a9b094a7d224cae2/tensorrt_llm/_torch/modules/fused_moe/routing.py#L255

File: tensorrt_llm/_torch/modules/fused_moe/routing.py [hunk 4]
```diff
@@ -86,18 +86,29 @@ class RenormalizeMoeRoutingMethod(BaseMoeRoutingMethod):
...
         top_k: int,
+        force_enable_pytorch_op: bool = False,
     ):
         super().__init__()
         self.top_k = top_k
+        self.force_enable_pytorch_op = force_enable_pytorch_op
 
-    def apply(self,
-              router_logits: torch.Tensor) -> (torch.Tensor, torch.Tensor):
+    def execute_pytorch_op(
```
[the remark this answers is not in the document]
Comment (ChristinaZ) [reply tensorrt_llm/_torch/modules/fused_moe/routing.py]: Done.

**Review by byshiue — APPROVED** [review]
